### PR TITLE
Semantic token provider

### PR DESCRIPTION
**Pull request by afflerbach**

### CODE REVIEW

File: src/checks.js [hunk 1]
```diff
@@ -152,6 +157,10 @@ function checkAttributeValue(name, value) {
 		return CheckFailed(invalidType)
 	} else if (REGEXES.boolean.test(value) && !types.includes("boolean")) {
 		return CheckFailed(invalidType)
+	} else if (REGEXES.function.test(value) && !functions.includes(RegExp.$1)) {
+		return CheckFailed(`Invalid function "${RegExp.$1}".`)
+	} else if (REGEXES.variable.test(value) && !variables.includes(RegExp.$1)) {
```
Comment (johakoch) [inline src/checks.js:162]:
```hcl
response {
  status = 401
```
is reported as 'Invalid variable "401"'.

File: src/checks.js [hunk 2]
```diff
@@ -152,6 +157,10 @@ function checkAttributeValue(name, value) {
 		return CheckFailed(invalidType)
 	} else if (REGEXES.boolean.test(value) && !types.includes("boolean")) {
 		return CheckFailed(invalidType)
+	} else if (REGEXES.function.test(value) && !functions.includes(RegExp.$1)) {
```
Comment (johakoch) [inline src/checks.js:160]:
```hcl
response {
    body = oauth2_verifierrrr()
```
is **not** reported.

**Review by johakoch — CHANGES_REQUESTED** [review]

File: src/checks.js (hunk 1 above)
Comment (afflerbach) [reply src/checks.js:162]: :+1:

File: src/checks.js (hunk 2 above)
Comment (afflerbach) [reply src/checks.js:160]: Cannot reproduce that one, though.

Comment (johakoch) [reply src/checks.js:160]: Me neither, hmm

**Review by johakoch — APPROVED** [review]

**Pull request merged**